app/src/main/java/org/p2p/wallet/newsend/SendFeeRelayerManager.kt

```diff
@@ -131,14 +132,17 @@ class SendFeeRelayerManager(
                 return
             }
 
-            val inputAmount = tokenAmount.toLamports(sourceToken.decimals)
             showFeeDetails(
                 sourceToken = sourceToken,
                 feeRelayerFee = feeRelayerFee,
                 feePayerToken = feePayer,
                 inputAmount = inputAmount,
                 strategy = strategy
             )
+        } catch (illegalStateException: IllegalStateException) {
+            Timber.i(illegalStateException, "Error during FeeRelayer fee calculation try to switch to sol")
+            sendInteractor.switchFeePayerToSol(solToken)
+            recalculate(sourceToken, inputAmount)
```

Comment:
I think it's a wrong place to handle that. 
There will be not only fee exceptions come here. 

You better catch it here: 

```
private suspend fun calculateFeeRelayerFee
```